File: V3/api_server/src/redis.ts
import { env } from "process";
import { createClient } from "redis";

let client: any
let subscriber: any
async function redisClient() {

  const redisUrl = env.REDIS_URL || "redis://localhost:6379"
  try {
    client = createClient({ url: redisUrl })
    subscriber = createClient({ url: redisUrl })

    await client.connect()
    await subscriber.connect()

    client.on("error",
      (err: Error) => console.error("Redis client error ", err)
    );
  } catch (error) {
    return console.error("Error conecting to redis", error)
  }
}

redisClient();

export { client, subscriber }

File: V3/auth_server/internal/redis/redis.go
package redis

import (
	"context"
	"fmt"
	"os"
	"time"

	"github.com/redis/go-redis/v9"
)

type Client struct {
	*redis.Client
}

func NewClient() (*Client, error) {
	redisAddr := os.Getenv("REDIS_ADDR")
	if redisAddr == "" {
		redisAddr = "localhost:6379"
	}

	rdb := redis.NewClient(&redis.Options{
		Addr: redisAddr,
	})

	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
	defer cancel()

	_, err := rdb.Ping(ctx).Result()
	if err != nil {
		return nil, fmt.Errorf("failed to connect to Redis: %w", err)
	}

	return &Client{rdb}, nil
}
